### DISCUSSION TITLE
Nanoels

### DISCUSSION THREAD

**Discussion by sikanderc21** [kachurovskiy/nanoels#108] [Show and tell]
Dear sir nanoels is very beautiful and sporting tool if you add two features more in it its as a supper controller for lathe machine one lcd screen use with I2C and external usb add for read gcode instructions for machining i hope its more difficult but not impossible for you thanks a lot my dear

**Comment by kachurovskiy** [comment]
@sikanderc21 I'd like to understand more about why you need an I2C screen support. We support I2C in the H1 version of the NanoEls - https://github.com/kachurovskiy/nanoels/blob/main/h1/NanoEls.ino#L111 - but in H2 this wasn't needed.

Did you build NanoEls on a breadboard? Maybe post a few photos of your build if you can.

As for gcode/grbl, sadly this is not possible with NanoEls but in general it's just a 2-axis CNC so perhaps something like https://howtomechatronics.com/tutorials/how-to-setup-grbl-control-cnc-machine-with-arduino/ could work for you.

Thank you.
